pipelines/api-pipelines.yml

```diff
@@ -100,29 +100,57 @@ stages:
               region: eastus
             centralus:
               region: centralus
-        steps:    
+        steps:
+          - task: UseDotNet@2
+            displayName: Dotnet version 3.1.100
+            inputs:
+              version: '3.1.100'
+
           - task: DownloadBuildArtifacts@0
             displayName: Download Build Artifacts
             inputs:
               buildType: 'current'
               downloadType: 'single'
               artifactName: 'drop'
               downloadPath: '$(Build.ArtifactStagingDirectory)'
+
           - task: AzureRMWebAppDeployment@4
-            displayName: Deploy to App Service
+            displayName: Deploy to Staging Slot
             inputs:
               ConnectionType: 'AzureRM'
               azureSubscription: $(azureSubscription)
               appType: 'apiApp'
+              ResourceGroupName: 'rg-$(businessUnit)-$(appName)-$(environment)-$(region)'
```

Comment:
minor - would we be able to have a step to populate variables with the RG name, backend endpoint...etc and reuse those variables through out the pipeline?

Reply:
This runs inside a parallel matrix build so we couldn't create independently before that.